src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c

```diff
@@ -1301,36 +1311,61 @@ int xocl_subdev_create_vsec_devs(xdev_handle_t xdev)
 	ret = xocl_subdev_vsec(xdev, XOCL_VSEC_FLASH_CONTROLER, &bar, &offset,
 		&vtype);
 	if (!ret) {
-		struct xocl_subdev_info subdev_info = XOCL_DEVINFO_FLASH_VSEC;
+		struct xocl_subdev_info subdev_info =
+		    XOCL_DEVINFO_FLASH_VSEC;
+
+		switch (vtype) {
+		case XOCL_VSEC_FLASH_TYPE_QSPI:
+			memcpy(((struct xocl_flash_privdata *)
+			    (subdev_info.priv_data))->flash_type,
+			    FLASH_TYPE_QSPIPS, strlen(FLASH_TYPE_QSPIPS));
+			/* default is FLASH_TYPE_SPI, thus pass through. */
```

Comment:
should we add __attribute__ ((fallthrough)); to suppress GCC warning?
https://github.com/Xilinx/XRT/pull/2405/files